test/core/end2end/fixtures/http_proxy_fixture.c

```diff
@@ -304,6 +307,45 @@ static void on_server_connect_done(grpc_exec_ctx* exec_ctx, void* arg,
                       &conn->on_write_response_done);
 }
 
+/**
+ * Parses the proxy auth header value to check if it matches :-
+ * Basic <base64_encoded_user_cred>
+ * Returns true if it matches, false otherwise
+ */
+static bool proxy_auth_header_matches(grpc_exec_ctx *exec_ctx,
+                                      char *proxy_auth_header_val) {
+  if(proxy_auth_header_val == NULL) {
+    return false;
+  }
+  char **auth_header_strs;
+  size_t auth_header_nstrs;
+  bool auth_header_matches = false;
+  // Split the auth header value on space
+  gpr_string_split(proxy_auth_header_val, " ", &auth_header_strs,
```

Comment:
In this particular case, I think `gpr_string_split()` is overkill, since we know what the first part of the string should be.  I suggest something like this instead:

```
if (strncmp(proxy_auth_header_val, "Basic ", 6) != 0) return false;
proxy_auth_header_val += 6;
gpc_slice decoded_slice =
    grpc_base64_decode(exec_ctx, proxy_auth_header_val, 0);
const bool match = grpc_slice_str_cmp(
    decoded_slice, GRPC_END2END_HTTP_PROXY_TEST_CONNECT_CRED) != 0;
grpc_slice_internal_unref(exec_ctx, decoded_slice);
return match;
```